### PR TITLE
Ogs without coupling

### PR DESCRIPTION
**Pull request by jbathmann**
This PR provides the possibility to use ogs to calculate belowground salinity distribution resulting from varying abiotic drivers. Here, no feedback from the plant water uptake to the porewater salinity distribution is regarded. So we could analyse the influence of the feedback on the porewater salinity and the plant distribution by comparison to other scenarios

### CODE REVIEW

File: TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/python_source.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import vtk as vtk
+import numpy as np
+from math import pi, sin
+import os
+
+seaward_salinity = 0.035
+tide_daily_amplitude = .7
+tide_monthly_amplitude = .37
+tide_daily_period = 60 * 60 * 12.
+tide_monthly_period = 60. * 60 * 24 * 31 / 2.
```
Comment (waltherm) [inline TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/python_source.py]: Do you intend to provide user editing possibility for these values at some time?

File: TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/OGSWithoutFeedback.py [hunk 2]
```diff
@@ -104,6 +116,7 @@ def copyPythonScript(self):
             if self._abiotic_drivers:
                 for abiotic_factor in self._abiotic_drivers.iterchildren():
                     if (abiotic_factor.tag + " = ") in line:
+
```
Comment (waltherm) [inline TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/OGSWithoutFeedback.py]: reason for line feed?

File: TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/python_source.py [hunk 3]
```diff
@@ -90,7 +91,7 @@ def getFlux(self, t, coords, primary_vars):
     "TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback")
 cumsum_savename = os.path.join(mangapath, "cumsum_salinity.npy")
 calls_savename = os.path.join(mangapath, "calls_in_last_timestep.npy")
-
+t_min = 1.5e7
```
Comment (waltherm) [inline TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/python_source.py]: why the hard coded value here?

File: TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/OGSWithoutFeedback.py [hunk 4]
```diff
@@ -19,7 +19,13 @@
...
     def __init__(self, args):
         super().__init__(args)
-        self.runOGSOnce()
+        if args.find("use_old_ogs_results"):
+            use_old_ogs_results = bool(args.find("use_old_ogs_results").text)
+            if use_old_ogs_results:
+                print("pyMANGA is using old results from previously saved"+
+                      " numpy arrays.")
+        else:
+            self.runOGSOnce()
```
Comment (waltherm) [inline TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/OGSWithoutFeedback.py]: I suppose this is thought to use previously simulated result, right? Hard to follow without an example on how to use it.

File: TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/python_source.py (hunk 1 above)
Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/python_source.py]: they are already implemented and documented. Please see https://github.com/jbathmann/pyMANGA/pull/27/files#diff-35cdd5495254eca5d8aa6c096713e0bd

File: TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/OGSWithoutFeedback.py (hunk 2 above)
Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/OGSWithoutFeedback.py]: fixed

File: TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/python_source.py (hunk 3 above)
Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/python_source.py]: you can change this via "abiotic_drivers" tag in the project file, if necessary. However, I intend to change the procedure such that users have to provider their own boundary condition scripts with just this example left, as I experience the need of large flexibility within the boundary conditions..

File: TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/OGSWithoutFeedback.py (hunk 4 above)
Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/OGSWithoutFeedback.py]: In the documentation I tried to explain that, if you have already calculated some salinity distribution for a given setup, it can be reused by setting the tag to "True".

https://github.com/jbathmann/pyMANGA/pull/27/files#diff-ff2e6c759d8ee737f30b4683e1cb83a8

File: TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/python_source.py (hunk 3 above)
Comment (waltherm) [reply TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/python_source.py]: True, if you want to run a sensitivity analysis, you will need to change such values a lot.

File: Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/OGSWithoutFeedback/s_use_old_ogs_results.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+
+
+Optional bool parameter. If this parameter is set to "True", pyMANGA is using the results on the porewater salinity distribution from its previous ogs-calculations.
```
Comment (waltherm) [inline Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/OGSWithoutFeedback/s_use_old_ogs_results.md]: Ok, this is belonging to one of my previous comments. I don't completely understand the meaning here, though. What is "previous ogs-calculations"? Any other former (even solitude) OGS result as a vtu file? Does this also mean, that it will actually never run OGS?

File: TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/OGSWithoutFeedback.py (hunk 4 above)
Comment (waltherm) [reply TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/OGSWithoutFeedback.py]: See comment in link above.

File: TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/python_source.py (hunk 1 above)
Comment (waltherm) [reply TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/python_source.py]: Ok, so these are only default values?

Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/OGSWithoutFeedback/python_source.py]: Yep.

File: Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/OGSWithoutFeedback/s_use_old_ogs_results.md (hunk 5 above)
Comment (jbathmann) [reply Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/OGSWithoutFeedback/s_use_old_ogs_results.md]: Maybe I could add something to the description as follows:

Imagine you want to do an parameter estimate for BETTINA model parameters. Here, you could want to have a salinity distribution resulting from certain abiotic conditions. In this case, the procedure would be: 
1.) Run pyMANGA with use_old_ogs_results = False to calculate the resulting salinity distribution.
2.) For each subsequent run of pyMANGA, you can use temporarily saved data on the subsurface salinity by setting use_old_ogs_results = True. Using this, the user could avoid redundant ogs calculations, because pyMANGA is using old results instead of recalculating the same salinity distribution again.

Comment (waltherm) [reply Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/OGSWithoutFeedback/s_use_old_ogs_results.md]: Ok, this is better. Maybe, Jonas can prepare an example that the specific usage becomes clear?

**Comment by waltherm** [comment]
:+1: from my side...
